app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsSecurityScreen.kt

```diff
@@ -96,6 +118,74 @@ object SettingsSecurityScreen : SearchableSettings {
                     .associateWith { stringResource(it.titleResId) },
             ),
             // SY -->
+            Preference.PreferenceItem.SwitchPreference(
+                pref = securityPreferences.passwordProtectDownloads(),
+                title = stringResource(R.string.password_protect_downloads),
+                subtitle = stringResource(R.string.password_protect_downloads_summary),
+                enabled = CbzCrypto.isPasswordSet(),
```

Comment:
All these should be gathered as a state above all the other preferences, maybe something like
```
val scope = rememberCoroutineScope()
val isCbzPasswordSet by remember {
    securityPreferences.cbzPassword().changes()
        .map { it.isNotEmpty() }
        .stateIn(scope, SharingStarted.Eagerly, false)
}.collectAsState()
```
Of course you can move some of the code to the CbzCrypto class, maybe a function that takes a CoroutineScope and returns a StateFlow that you can use .collectAsState() on.

Reply:
I changed it